syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.mail";
option java_outer_classname = "MailProto";
option objc_class_prefix = "MAIL";

package mail;

service Mail {

  rpc GetMailList (GetMailListRequest) returns (GetMailListReply) {}
  rpc GetMail (GetMailRequest) returns (GetMailReply) {}
  rpc PutMail (PutMailRequest) returns (PutMailReply) {}
  
}

message GetMailListRequest {
  string user = 1;
}

message GetMailListReply {
	repeated Email item = 1;
}

message Email {
  string from = 1;
  string subject = 2;
  string date = 3;
}

message GetMailRequest {
  string user = 1;
  string email_id = 2;
}

message GetMailReply {
	string content = 1;
}

message PutMailRequest {
  string receiver = 1;
  string created_time = 2;
  string subject = 3;
  string sender = 4; 
  string content = 5;
  int32 size = 6;
}

message PutMailReply {
  string email_id = 1;
}
